### PR TITLE
Add ultrawarm support and advanced security features

### PR DESCRIPTION
**Pull request by maciejasembler**
## what
* Adds in-module support for ultrawarm nodes
* Adds in-module support for advanced security features (fine-grained access control)

## why
* People are using it, obviously

## references
* https://github.com/cloudposse/terraform-aws-elasticsearch/issues/69

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -118,6 +118,24 @@ variable "instance_count" {
...
 
+variable "warm_enabled" {
+  type        = bool
+  default     = false
+  description = "Whether AWS UltraWarm is enabled"
+}
+
+variable "warm_count" {
+  type        = number
+  default     = 2
```
Comment (dekimsey) [inline variables.tf:129]: I'm wondering, should we even bother with a `warm_enabled` flag? Could we just state warm_count > 0? I'm wondering if the module should hide the relationship between the two settings or act as a very thin shim and expose it.

> warm_enabled - (Optional) Indicates whether to enable warm storage.
> warm_count - (Optional) The number of warm nodes in the cluster. Valid values are between 2 and 150. warm_count can be only and must be set when warm_enabled is set to true.

Comment (aknysh) [reply variables.tf:129]: @deiga we usually provide all the variables that a resources has, so having both in this case is ok
please rebase, and thanks for the PR

**Comment by aknysh** [comment]
@maciejasembler please rebase

**Comment by maciejasembler** [comment]
On it.

**Comment by maciejasembler** [comment]
Messed this branch so I opened new PR:
https://github.com/cloudposse/terraform-aws-elasticsearch/pull/74

**Pull request closed without merging**